### PR TITLE
create the ReservationService

### PR DESCRIPTION
**Pull request by Beykn**

### CODE REVIEW

File: main-service/src/service/ReservationService.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package service;
+
```
Comment (JamshidS) [inline main-service/src/service/ReservationService.java:2]: this is wrong you  should create a interface after that there should be a concrete class.

**Review by JamshidS — CHANGES_REQUESTED** [review]

File: main-service/src/service/ReservationService.java (hunk 1 above)
Comment (Beykn) [reply main-service/src/service/ReservationService.java:2]: I thought if we use the repository I should the create interface after concert a class like Reservation Service interface and Reservation ServiceImpl.

File: main-service/src/service/Impl/ReservationServiceImp.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.List;
+
+public class ReservationServiceImp implements ReservationService {
+
+    private List<Reservation> reservations = new ArrayList<>();
+    private long nextReservationId = 1;
+
+    //add a new reservation
+    @Override
+    public void addReservation(Reservation reservation){
```
Comment (JamshidS) [inline main-service/src/service/Impl/ReservationServiceImp.java:16]: Return a proper response.

File: main-service/src/service/Impl/ReservationServiceImp.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    //delete
+    @Override
+    public void deleteReservationById(long reservationId) {
+        reservations.removeIf(reservation -> reservation.getReservationId() == reservationId);
+    }
+
+    //this for save the original list , returns the copied one
+    @Override
+    public List<Reservation> getAllReservations() {
+        return new ArrayList<>(reservations);
```
Comment (JamshidS) [inline main-service/src/service/Impl/ReservationServiceImp.java:56]: ```return reservations;```

**Review by JamshidS — CHANGES_REQUESTED** [review]

**Pull request merged**